pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go

```diff
@@ -179,24 +218,30 @@ func (m *rangefeedMuxer) startSingleRangeFeed(
 		startAfter:      startAfter,
 		catchupRes:      catchupRes,
 		token:           token,
+		retry:           retry.StartWithCtx(ctx, m.ds.rpcRetryOptions),
 	}
-	streamID := atomic.AddInt64(&m.seqID, 1)
 
-	// stream ownership gets transferred (indicated by setting stream to nil) when
-	// we successfully send request. If we fail to do so, cleanup.
-	defer func() {
-		if stream != nil {
-			stream.release()
-		}
-	}()
+	if err := stream.start(ctx, m); err != nil {
+		stream.release()
+		return err
+	}
 
-	// Start a retry loop for sending the batch to the range.
-	for r := retry.StartWithCtx(ctx, m.ds.rpcRetryOptions); r.Next(); {
-		maybeReleaseTransport()
+	return nil
+}
 
+// start begins execution of activeMuxRangeFeed.
+// Backoff is applied if same stream restarts.
+func (s *activeMuxRangeFeed) start(ctx context.Context, m *rangefeedMuxer) error {
+	streamID := atomic.AddInt64(&m.seqID, 1)
+
+	// Start a retry loop for sending the batch to the range.
+	for s.retry.Next() {
```

Comment:
Carrying the `transport`, `token` and friends seems beneficial.

I'm not sure that reusing the `retry` state across starts is great, if I understand this implementation. Say, `start` succeeds, and we run the `activeMuxRangeFeed` for a while. Then a transient error happens during the event handling loop, we try to restart the feed. Why would we use the same `retry` state that we used to start the feed in the first place?

The delay in `retry` will keep growing. So, with every transient error we will be starting the feed slower and slower. I think we should sometimes (when things are going smoothly again) decrease the delay, or reset the state of `retry`. For example, if the time after the last successful start is more than a threshold, then we `s.retry.Reset()` here.

Reply:
It's not that bad since the backoff is bound by 1 sec; but I was thinking the same, just wanted to keep things
as close to existing impl as possible.  Having said that, since you're voicing the same concern I had, I removed retry state from the activeMux.  No need to reset it.  Basically, if you were able to establish node connection, plus send an initial request, then you've done enough work -- and retry can happen immediately.

Verified this fix is fine with decommission test.